larky/src/main/java/com/verygood/security/larky/modules/vgs/metrics/MetricsModule.java

```diff
@@ -56,65 +60,118 @@ public MetricsModule() {
         name = "amount",
         named = true,
         doc = "Amount",
+        positional = false,
         allowedTypes = {
-          @ParamType(type = Integer.class)
+          @ParamType(type = StarlarkInt.class),
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
         }
       ),
       @Param(
         name = "bin",
         named = true,
+        positional = false,
         doc = "Bank Identification Number",
         allowedTypes = {
-          @ParamType(type = Integer.class)
+          @ParamType(type = StarlarkInt.class),
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
         }
       ),
       @Param(
         name = "currency",
         named = true,
+        positional = false,
         doc = "Currency",
-        defaultValue = "NOT_SPECIFIED",
         allowedTypes = {
-          @ParamType(type = Currency.class)
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
         }
       ),
       @Param(
         name = "psp",
         named = true,
+        positional = false,
         doc = "Payment Service Provider",
-        defaultValue = "NOT_SPECIFIED",
         allowedTypes = {
-          @ParamType(type = PSP.class)
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
         }
       ),
       @Param(
         name = "result",
         named = true,
+        positional = false,
         doc = "Transaction Result",
-        defaultValue = "NOT_SPECIFIED",
         allowedTypes = {
-          @ParamType(type = TransactionResult.class)
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
         }
       ),
       @Param(
         name = "type",
         named = true,
+        positional = false,
         doc = "Transaction Type",
-        defaultValue = "NOT_SPECIFIED",
         allowedTypes = {
-          @ParamType(type = TransactionType.class)
+          @ParamType(type = String.class),
+          @ParamType(type = NoneType.class),
+        }
+      ),
+      @Param(
+        name = "dictionary",
```

Comment:
Still don't really understand the value of this

Reply:
https://github.com/verygoodsecurity/starlarky/pull/458/files#diff-9a67394f942f4e6f04f7eac1196aec6572dd066a0967600581a42ded69d26b46R14